src/main/java/Control/Scenario/UC_FlatSummary.java

```diff
@@ -0,0 +1,175 @@
+package Control.Scenario;
+
+import Control.Connect.DbProvider;
+import Model.Database.Interaction.ComplexInteractions.GeneralAccessibility;
+import Model.Database.Interaction.*;
+import Model.Database.Tables.Table.*;
+import Model.Web.Project;
+import Model.Web.Projects;
+
+import javax.validation.constraints.NotNull;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+/**
+ * Use Case class for FlatSummaryView
+ */
+public class UC_FlatSummary {
+    private DbProvider db;
+
+    public UC_FlatSummary(@NotNull DbProvider dbProvider) {
+        this.db = dbProvider;
+    }
+
+    public final Projects allProjectsForUser(int userID) {
+        Projects projects = new Projects();
+
+        try {
+            db.beforeSqlExecution();
+
+            ArrayList<T_Project_user> arr = I_ProjectUser.retrieveAllForUser(db.getConn(), db.getPs(), db.getRs(), userID);
+            ArrayList<Project> temp = new ArrayList<Project>();
+
+            for (T_Project_user  tpu: arr) {
+                Project project = new Project();
+
+                T_Project tp = I_Project.retrieve(db.getConn(), db.getPs(), db.getRs(), tpu.getA_ProjectID());
+
+                project.setId(tp.getA_pk());
+                project.setName(tp.getA_Name());
+                project.setCreatedat(tp.getA_CreatedAt());
+                project.setDeletedat(tp.getA_DeletedAt());
+
+                temp.add(project);
+            }
+
+            db.afterOkSqlExecution();
+
+            // dont forget to set data that was inserted into json
+            projects.setProjects(temp);
+
+        } catch (SQLException sqle) {
+            db.afterExceptionInSqlExecution(sqle);
+        }
+
+        return projects;
+    }
+
+    /***
+     * Returns null if he has no right to see measurements for that flat or it does not exist
+     * @param userId from sesssion
+     * @param flatId from request
+     * @return
+     */
+    public ArrayList<T_Measurement> getAllMeasurementsForFlat(int userId, int flatId) {
```

Comment:
ked to dokncime musime sa dohodnut na nejakej konvencii. Napr tam boli metody vsade retrieve, tak som sa snazim pouzivat to a nie get, malo by to byt jednotne, hlavne ked tam uz potom bude 100 roznych volani v kontroleri

Reply:
nie su metody retrieve len ked tahas data SQLkami z databazy? ze ked robis uz nejaku vyssiu logiku tak budeme davat get? ale suhlasim treba dohodnut nejaku konvenciu nakolko toto je example podla ktoreho sa budu robit ostatne

Reply:
to by bolo fajn, že retrieve z databazy a get na higher logic nech je to na prvy pohlad znat